dashboard/modules/job/job_manager.py

```diff
@@ -71,14 +74,26 @@ def tail_logs(self, job_id: str) -> Iterator[str]:
     def get_last_n_log_lines(
         self, job_id: str, num_log_lines=NUM_LOG_LINES_ON_ERROR
     ) -> str:
+        """
+        Returns the first MAX_LOG_SIZE (20000) characters for the last
+        `num_log_lines` lines.
+
+        Args:
+            job_id: The id of the job whose logs we want to return
+            num_log_lines: The number of lines to return.
+        """
         log_tail_iter = self.tail_logs(job_id)
         log_tail_deque = deque(maxlen=num_log_lines)
-        for line in log_tail_iter:
-            if line is None:
+        for lines in log_tail_iter:
+            if lines is None:
                 break
             else:
-                log_tail_deque.append(line)
-        return "".join(log_tail_deque)
+                # log_tail_iter can return batches of lines at a time.
+                # Need to split into lines
+                for line in lines.splitlines():
+                    log_tail_deque.append(line)
+
+        return "\n".join(log_tail_deque)[0 : self.MAX_LOG_SIZE]
```

Comment:
could we return the **last** max characters of the last lines? (the most important error message at the end will be neglected if the lines contain >> 20000 chars)